superset-frontend/plugins/plugin-chart-echarts/src/utils/series.ts

```diff
@@ -155,6 +151,84 @@ export function sortAndFilterSeries(
   ).map(({ name }) => name);
 }
 
+export function sortRows(
+  rows: DataRecord[],
+  xAxis: string,
+  xAxisSortSeries: SortSeriesType,
+  xAxisSortSeriesAscending: boolean,
+) {
```

Comment:
I originally set out to implement this with `lodash`, similar to how `sortAndFilterSeries` was implemented, but in this case it felt clearer to write this out like this. But if this function looks too explicit I can rewrite this with the equivalent `lodash` functions (refactoring should be simple, as the tests should work as-is).